src/main/java/org/apache/pirk/encryption/Paillier.java

```diff
@@ -95,140 +95,185 @@
-   * to the value of this parameter.
-   *
+   * The probability that the generated keys represent primes will exceed (1 - (1/2)<sup>{@code certainty}</sup>). The execution time of this constructor is
+   * proportional to the value of this parameter.
+   * <p>
+   * When ensureBitSet > -1 the value of bit "{@code ensureBitSet}" in modulus {@code N} will be set.
+   * 
+   * @param bitLength
+   *          The bit length of the resulting modulus {@code N}.
+   * @param certainty
+   *          The probability that the new {@code p} and {@code q} represent prime numbers.
+   * @param ensureBitSet
+   *          index of bit in {@code N} to ensure is set.
+   * @throws IllegalArgumentException
+   *           If the {@code certainty} is less than the system allowed lower bound, or the index of {@code ensureBitSet} is greater than the {@code bitLength}.
    */
-  public Paillier(int bitLengthInput, int certainty, int ensureBitSet) throws PIRException
+  public Paillier(int bitLength, int certainty, int ensureBitSet)
   {
-    bitLength = bitLengthInput;
-
     int systemPrimeCertainty = SystemConfiguration.getIntProperty("pir.primeCertainty", 128);
     if (certainty < systemPrimeCertainty)
     {
-      throw new PIRException("Input certainty = " + certainty + " is less than allowed system lower bound = " + systemPrimeCertainty);
+      throw new IllegalArgumentException("Input certainty = " + certainty + " is less than allowed system lower bound = " + systemPrimeCertainty);
     }
-    if (ensureBitSet >= bitLengthInput)
+    if (ensureBitSet >= bitLength)
     {
-      throw new PIRException("ensureBitSet = " + ensureBitSet + " must be less than bitLengthInput = " + bitLengthInput);
+      throw new IllegalArgumentException("ensureBitSet = " + ensureBitSet + " must be less than bitLengthInput = " + bitLength);
     }
-    generateKeys(certainty, ensureBitSet);
+    this.bitLength = bitLength;
+    generateKeys(bitLength, certainty, ensureBitSet);
     setDerivativeElements();
 
     logger.info("Parameters = " + parametersToString());
   }
 
+  /**
+   * Returns the value of the large prime {@code p}.
+   * 
+   * @return p.
+   */
   public BigInteger getP()
   {
     return p;
   }
 
+  /**
+   * Returns the value of the large prime {@code q}.
+   * 
+   * @return q.
+   */
   public BigInteger getQ()
   {
     return q;
   }
 
+  /**
+   * Returns the RSA modulus value {@code N}.
+   * 
+   * @return N, the product of {@code p} and {@code q}.
+   */
   public BigInteger getN()
   {
     return N;
   }
 
+  /**
+   * Returns the value of {@code N}<sup>2</sup>.
+   * 
+   * @return N squared.
+   */
   public BigInteger getNSquared()
   {
     return NSquared;
   }
 
+  /**
+   * Returns the value of Carmichael's function at {@code N}.
+   * <p>
+   * The Carmichael function of {@code N} is the lowest common multiple of {@code p-1} and {@code q-1},
```

Comment:
LCM = Least Common Multiple (maybe 'lowest common multiple' is British terminology? ;))